src/main/java/io/pravega/connectors/flink/EventTimeOrderingFunction.java

```diff
@@ -0,0 +1,125 @@
+/**
+ * Copyright (c) Dell Inc., or its subsidiaries. All Rights Reserved.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.connectors.flink;
+
+import org.apache.flink.api.common.state.MapState;
+import org.apache.flink.api.common.state.MapStateDescriptor;
+import org.apache.flink.api.common.state.ValueState;
+import org.apache.flink.api.common.state.ValueStateDescriptor;
+import org.apache.flink.api.common.typeinfo.BasicTypeInfo;
+import org.apache.flink.api.common.typeinfo.TypeInformation;
+import org.apache.flink.api.java.typeutils.ListTypeInfo;
+import org.apache.flink.configuration.Configuration;
+import org.apache.flink.streaming.api.functions.KeyedProcessFunction;
+import org.apache.flink.util.Collector;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class EventTimeOrderingFunction<T> extends KeyedProcessFunction<String, T, T> {
+
+    private static final long serialVersionUID = 1L;
+
+    private static final String EVENT_QUEUE_STATE_NAME = "eventQueue";
+
+    private static final String LAST_TRIGGERING_TS_STATE_NAME = "lastTriggeringTsState";
+
+    /**
+     * The input type information for buffering events to managed state.
+     */
+    private final TypeInformation<T> typeInformation;
+
+    /**
+     * The queue of input elements keyed by event timestamp.
+     */
+    private transient MapState<Long, List<T>> elementQueueState;
+
+    /**
+     * State to keep the last triggering timestamp. Used to filter late events.
+     */
+    private transient ValueState<Long> lastTriggeringTsState;
+
+    public EventTimeOrderingFunction(TypeInformation<T> typeInformation) {
+        this.typeInformation = typeInformation;
+    }
+
+    @Override
+    public void open(Configuration config) throws Exception {
+        super.open(config);
+
+        // create a map-based queue to buffer input elements
+        if (elementQueueState == null) {
```

Comment:
It is redundant to have the null check here, we can always get the state from the context.

Reply:
remove the check and create the map whenever the `open` is called